test/packages/file_upload_example/file_upload/files_data/fields/fields.yml

```diff
@@ -0,0 +1,180 @@
+      description: file created time
+    - name: device
+      type: keyword
+      description: device
+    - name: drive_letter
+      type: keyword
+      description: drive letter
+    - name: extension
+      type: keyword
+      description: file extension
+    - name: directory
+      type: keyword
+      description: file directory
+    - name: gid
+      type: keyword
+      description: gid
+    - name: group
+      type: keyword
+      description: group
+    - name: inode
+      type: keyword
+      description: inode
+    - name: mode
+      type: keyword
+      description: mode
+    - name: mtime
+      type: date
+      description: mtime
+    - name: owner
+      type: keyword
+      description: file owner
+    - name: path
+      type: keyword
+      description: file path
+    - name: target_path
+      type: keyword
+      description: target path
+    - name: type
+      type: keyword
+      description: file type
+    - name: uid
+      type: keyword
+      description: uid
+- name: contents
+  title: Contents
+  type: group
+  fields:
+        - name: size
+      type: long
+      description: file size
+    - name: name
+      type: keyword
+      description: file name
+    - name: mime_type
+      type: keyword
+      description: file mime type
+    - name: hash
+      title: Hash
+      type: group
+      fields:
+        - name: sha256
+          type: keyword
+          description: file hash in sha256
+    - name: accessed
+      type: date
+      description: date accessed
+    - name: attributes
+      type: keyword
+      description: file attributes
+    - name: created
+      type: date
+      description: file created date
+    - name: ctime
+      type: date
+      description: file created time
+    - name: device
+      type: keyword
+      description: device
+    - name: drive_letter
+      type: keyword
+      description: drive letter
+    - name: extension
+      type: keyword
+      description: file extension
+    - name: directory
+      type: keyword
+      description: file directory
+    - name: gid
+      type: keyword
+      description: gid
+    - name: group
+      type: keyword
+      description: group
+    - name: inode
+      type: keyword
+      description: inode
+    - name: mode
+      type: keyword
+      description: mode
+    - name: mtime
+      type: date
+      description: mtime
+    - name: owner
+      type: keyword
+      description: file owner
+    - name: path
+      type: keyword
+      description: file path
+    - name: target_path
+      type: keyword
+      description: target path
+    - name: type
+      type: keyword
+      description: file type
+    - name: uid
+      type: keyword
+      description: uid
+- name: event
+  title: Event
+  type: group
+  fields:
+    - name: id
+      type: keyword
+      description: event id
+- name: host
+  title: Host
+  type: group
+  fields:
+    - name: hostname
+      type: keyword
+      description: hostname
```

Comment:
These mappings look to me like quite generic files metadata. How will these mappings be different in other packages? Do you have an example?

Reply:
There is a mandatory minimum set of fields for the functionality to work.

I believe the set is currently (but not finalized):

- `file.Size`
- `file.Status`
- `file.ChunkSize`
- `file.Name`


Any integration (endpoint, agent, or future users) should be free to map any additional properties that might aid in:

- displaying the content (e.g. mime type, file path)
- filtering and discovery (e.g. event.id, owner, permissions)
- investigation and information (e.g. inode, owner, create time, access times)

Importantly, an integration is responsible for its own discovery and filtering of files as desired. This index mapping directly feeds whatever is available for search and filtering. What Endpoint may want to index for searching could be different from Agent. And I can see Endpoint's mapping expanding as time goes on, as security information about a quarantined file. It is unlikely these expanded uses would be _as_ useful to Agent who are uploading diagnostic logs for the time being. It is even less likely those additional fields need to be explicitly indexed rather than just _present in source doc_, if sent at all.


Other integrations, if this feature does end up useful in the future to others, will have their own needs to filtering and file discovery and this mapping would control that. I would posit that it needs to be flexible (with the few feature-required minimums)